mac_cleanup/core.py

```diff
@@ -208,28 +208,41 @@ def _count_dry(self) -> float:
         # Extracts paths from path_modules list
         path_list: list[Path_] = [path.get_path for path in path_modules]
 
-        # Set counter for estimated size
-        estimate_size: float = 0
-
         # Get thread executor
         executor = ThreadPoolExecutor()
 
         try:
             # Add tasks to executor
             tasks = [executor.submit(self._get_size, path) for path in path_list]
 
+            # Store paths and their corresponding futures
+            path_future_map = dict(zip(path_list, tasks))
+
             # Wait for task completion and add ProgressBar
             for future in ProgressBar.wrap_iter(
                 as_completed(tasks), description="Collecting dry run", total=len(path_list)
             ):
-                estimate_size += future.result(timeout=10)
+                path = [p for p, f in path_future_map.items() if f == future][0]
```

Comment:
maybe just a list and next call instead of a getter?
```suggestion
                path = next(p for p, f in path_future_zip if f == future)
```